crates/ruff_source_file/src/lib.rs

```diff
@@ -165,6 +165,7 @@ impl SourceFileBuilder {
 /// Cloning a [`SourceFile`] is cheap, because it only requires bumping a reference count.
 #[derive(Clone, Eq, PartialEq, Hash)]
 #[cfg_attr(feature = "get-size", derive(get_size2::GetSize))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
I'd prefer if we don't add serialize implementation for those types. We could simply panic in the `UnifiedFile` implementation whenever we encounter a ruff file.

The reason is that it seems unnecessary to maintain this complexity when Ruff doesn't even use salsa's caching and serializing the line index is a little useles (it's very cheap to build it from the source)